hookman/hooks.py

```diff
@@ -101,22 +103,28 @@ def plugins_available(self) -> Optional[List[PluginInfo]]:
         (configuration available at the YAML file and computed values )
         """
         plugin_config_files = hookman_utils.find_config_files(self.plugins_dirs)
-        return [PluginInfo(plugin_file) for plugin_file in plugin_config_files]
+        return [PluginInfo(plugin_file, self.hooks_available) for plugin_file in plugin_config_files]
 
     def get_hook_caller(self):
         """
-        Return a HookCaller class that holds all references for the functions implemented on the plugins.
+        Return a HookCaller class that holds all references for the functions implemented
+        on the plugins.
         """
-        _hookman = __import__(self.specs.pyd_name)
-        hook_caller = _hookman.HookCaller()
-        for plugin in self.plugins_available():
-            self._bind_libs_functions_on_hook_caller(plugin.shared_lib_path, hook_caller)
+        if not self.plugins_has_conflicts():
+            _hookman = __import__(self.specs.pyd_name)
+            hook_caller = _hookman.HookCaller()
+            for plugin in self.plugins_available():
+                self._bind_libs_functions_on_hook_caller(plugin.shared_lib_path, hook_caller)
```

Comment:
Slightly better to move the `return` call inside this `if` block because `hook_caller` is defined here.

Reply:
-[x] Move the return statement to inside the if block